src/validationSchemas/constraints.ts

```diff
@@ -0,0 +1,3 @@
+import joi from 'joi';
+
+export const constraintDateTypeSchema = joi.date();
```

Comment:
There is value in having the same schema validator on the frontend and the backend, it would be unfortunate if the backend rejects something that the frontend allows. But given the size of this library, and the fact that this is the only place we use it - I think you're correct here and this is a much larger discussion if we want to make this investment. I will defer to date-fns and remove joi for now.